telegram/request/_requestparameter.py

```diff
@@ -64,52 +64,76 @@ class RequestParameter:
 
     @property
     def multipart_data(self) -> Optional[UploadFileDict]:
         """A dict with the file data to upload, if any."""
         if not self.input_files:
             return None
-        return {input_file.attach_name: input_file.field_tuple for input_file in self.input_files}
+        return {
+            (input_file.attach_name or self.name): input_file.field_tuple
+            for input_file in self.input_files
+        }
 
     @staticmethod
-    def _value_and_input_file_from_input(  # pylint: disable=too-many-return-statements
+    def _value_and_input_files_from_input(  # pylint: disable=too-many-return-statements
         value: object,
     ) -> Tuple[object, List[InputFile]]:
-        """Converts `value` into something that we can json-dump. If `value` contains a file to be
-        uploaded, it will be returned as second return value and the corresponding attach:// value
-        will be returned as first return value.
-        Note that we use this for *all* files to be uploaded. This is not documented in the
-        official API, but has been confirmed to be supported in the official Bot API repository.
-        See https://github.com/tdlib/telegram-bot-api/issues/167
-
-        This method does no special casing for enums because
-        * all enums in tg.constants are subclasses of int/str, so they are already json-dumpable
+        """Converts `value` into something that we can json-dump. Returns two values:
+        1. the JSON-dumpable value. Maybe be `None` in case the value is an InputFile which must
+           not be uploaded via an attach:// URI
+        2. A list of InputFiles that should be uploaded for this value
+
+        Note that we handle files differently depending on whether attaching them via an URI of the
+        form attach://<name> is documented to be allowed or not.
+        There was some confusion whether this worked for all files, so that we stick to the
+        documented ways for now.
+        See https://github.com/tdlib/telegram-bot-api/issues/167 and
+        https://github.com/tdlib/telegram-bot-api/issues/259
+
+        This method only does some special casing for our own helper class StringEnum, but not
+        for general enums. This is because:
+        * tg.constants currently only uses IntEnum as second enum type and json dumping that
+          is no problem
         * if a user passes a custom enum, it's unlikely that we can actually properly handle it
           even with some special casing.
         """
         if isinstance(value, datetime):
             return to_timestamp(value), []
-        if isinstance(value, Enum):
+        if isinstance(value, StringEnum):
             return value.value, []
         if isinstance(value, InputFile):
-            return value.attach_uri, [
-                value,
-            ]
+            if value.attach_uri:
+                return value.attach_uri, [
+                    value,
+                ]
```

Comment:
did black make it like this

Reply:
yes, but removing the `,` after `value` does the trick to make it a single line :)